### ISSUE TITLE
[Feature Request] Hide metadata that is empty. Support Obsidian/Evernote links

### ISSUE THREAD

**Issue by dinhngocvuong** [arnau/obsidian-metatable#5]
Hi! First of all thank you for this plugin! I have 3 requests:

1. Could you add an option to hide metadate that is empty
![](https://i.imgur.com/ygc34yA.png)

2. Suport Obsidian, Evernote links... Set the max length of URL
![](https://i.imgur.com/kGYp17R.png)

3. When I click the tag, the windows popup always open instead of search
![](https://i.imgur.com/97JKB45.png)

Thank you!

**Comment by arnau** [comment]
Hi @dinhngocvuong, thanks for all the feedback!

> Could you add an option to hide metadate that is empty

Next release will include a toggle in settings to ignore members with null (empty) values.

> Suport Obsidian, Evernote links... Set the max length of URL

I need to look into this. I'm not familiar with Evernote or any obsidian plugin that handles this kind of plugin. Do you have any pointers? It might be a similar issue I have with #3 

> When I click the tag, the windows popup always open instead of search

**ugh** that's not good. Unfortunately I don't have access to any Windows machine to test this behaviour.

I did a change (next release) in hope it will fix it but it's a blind shot :S

**Comment by dinhngocvuong** [comment]
I just upgraded to v0.7.0 and see most my above issues were fixed. Thanks a lot!

The Evernote link is similar to Obsidian link on #3 

1 more question: How can I change the tag style like ![Imgur](https://i.imgur.com/9UxON6v.png)
I cannot override your plugin style

And I prefer the sign "#" before the tag name.

**Comment by arnau** [comment]
Glad to hear!

I'm still looking into the evernote ones, should be straightforward but I don't want to add something without testing at all :)

At the moment you can't change the tag styles I'm afraid (nor add the `#` in front of it), it's one of the consequences of having strong boundaries for the plugin. That said, I'm thinking in how could it be done though

**Comment by arnau** [comment]
Out of curiosity, what custom code do you use to have tags with different colours?

**Comment by dinhngocvuong** [comment]
You can check it out
https://forum.obsidian.md/t/meta-post-common-css-hacks/1978/13

**Comment by arnau** [comment]
So, the latest version should create links for Evernote and a custom property so you can add your own prefix for tags. E.g.

```css
# <your vault>/.obsidian/snippets/metatable.css
.obsidian-metatable {
  --metatable-tag-symbol: "#";
}
```

**Comment by dinhngocvuong** [comment]
Great! Thank you for your help.

![](https://i.imgur.com/WGbifk4.png)

**Comment by dinhngocvuong** [comment]
I always start my note template with metadata, but some quick notes don't need the metadata, and I want to hide the text "Metadata" on preview page when there is nothing to show.

![](https://i.imgur.com/hKRMvC6.png)

**Comment by arnau** [comment]
Umm, I fixed this case around version 0.8. I can certainly not reproduce it now. I'm testing with a clean obsidian vault with metatable 0.8.1 installed and nothing else.

Could you give me some details of version of Obsidian and metatable version please?

**Comment by dinhngocvuong** [comment]
I use Obsidian 0.12.3 for Windows 10, Metatable 0.8.1

Code view
![](https://i.imgur.com/flIDHjD.png)

Preview view
![](https://i.imgur.com/8lzlfop.png)

Setting
![](https://i.imgur.com/M034RaQ.png)

**Comment by arnau** [comment]
Ah, thanks. Will look into this case soon

**Comment by arnau** [comment]
Ok, version 0.8.2 fixes this case where all top-level keys are null 👍

**Comment by arnau** [comment]
Something to note though is that it won't work if you have nested nulls, e.g.:

```yaml
top:
  child:
```

Because `top` is technically not null even though that after pruning `child` because it's null, then `top` becomes null.

For now I'll keep it as a no-fix mainly because it would require a fairly expensive rework.

**Comment by dinhngocvuong** [comment]
Could you please fix for the case: Ignored keys
![](https://i.imgur.com/1MbpeWY.png)

In my case, I put `cssclass` to Ignored keys

Code view
![](https://i.imgur.com/wnwDZAR.png)

Preview view
![](https://i.imgur.com/PoF9M6U.png)

**Comment by arnau** [comment]
Fixed 😅 in 0.8.3 (when GitHub Actions comes back to live…)

**Comment by dinhngocvuong** [comment]
Thank you for the quick fix! 👍
